lib/htmlnano.es6

```diff
@@ -19,8 +23,57 @@ function htmlnano(options = {}, preset = safePreset) {
                 throw new Error('Module "' + moduleName + '" is not defined');
             }
 
-            let module = require('./modules/' + moduleName);
-            promise = promise.then(tree => module.default(tree, options, moduleOptions));
+            const module = require('./modules/' + moduleName);
+
+            if (module.once) {
```

Comment:
I don't quite understand the difference between traditional and _"once"_ modules.

Reply:
`once` module means the module is "independent": `on*` method only requires information from the `node` itself.

Some modules (`mergeScripts`, `removeOptionalTags`) requires "context" (the node before and after, the parent node, etc.), thus it is not "independent" and can not be refactored to `once` module

Reply:
`node`, `forEach`, `zerocontext`, `nocontext`, `nodeprocessor`, `strictnode`, `node-only`, `perNode`, `every`